utils.go

```diff
@@ -184,26 +143,27 @@ func newServicerAndStorager(pairs ...typ.Pair) (srv *Service, store *Storage, er
 
 // All available storage classes are listed here.
 const (
-	StorageClassStandard           = s3.ObjectStorageClassStandard
-	StorageClassReducedRedundancy  = s3.ObjectStorageClassReducedRedundancy
-	StorageClassGlacier            = s3.ObjectStorageClassGlacier
-	StorageClassStandardIa         = s3.ObjectStorageClassStandardIa
-	StorageClassOnezoneIa          = s3.ObjectStorageClassOnezoneIa
-	StorageClassIntelligentTiering = s3.ObjectStorageClassIntelligentTiering
-	StorageClassDeepArchive        = s3.ObjectStorageClassDeepArchive
+	StorageClassStandard           = types.ObjectStorageClassStandard
+	StorageClassReducedRedundancy  = types.ObjectStorageClassReducedRedundancy
+	StorageClassGlacier            = types.ObjectStorageClassGlacier
+	StorageClassStandardIa         = types.ObjectStorageClassStandardIa
+	StorageClassOnezoneIa          = types.ObjectStorageClassOnezoneIa
+	StorageClassIntelligentTiering = types.ObjectStorageClassIntelligentTiering
+	StorageClassDeepArchive        = types.ObjectStorageClassDeepArchive
 )
 
 func formatError(err error) error {
 	if _, ok := err.(services.InternalError); ok {
 		return err
 	}
 
-	e, ok := err.(awserr.RequestFailure)
-	if !ok {
+	fmt.Printf("%T", err)
+	e := &awshttp.ResponseError{}
+	if ok := errors.As(err, &e); !ok {
 		return fmt.Errorf("%w: %v", services.ErrUnexpected, err)
 	}
 
-	switch e.Code() {
+	switch err.Error() {
```

Comment:
First,  `&awshttp.ResponseError{}` maybe too gerenal to check the error code, let's use `&smithy.GenericAPIError` instead, so that we can check via `e.ErrorCode()`.

Then, we should not use `err` here anymore.

Reply:
Got it